presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorSplit.java

```diff
@@ -48,40 +50,75 @@
     public RaptorSplit(
             @JsonProperty("connectorId") String connectorId,
             @JsonProperty("shardUuids") Set<UUID> shardUuids,
+            @JsonProperty("shardDeltaMap") Map<UUID, UUID> shardDeltaMap,
+            @JsonProperty("tableSupportsDeltaDelete") boolean tableSupportsDeltaDelete,
             @JsonProperty("bucketNumber") OptionalInt bucketNumber,
             @JsonProperty("effectivePredicate") TupleDomain<RaptorColumnHandle> effectivePredicate,
             @JsonProperty("transactionId") OptionalLong transactionId,
             @JsonProperty("columnTypes") Optional<Map<String, Type>> columnTypes)
     {
-        this(connectorId, shardUuids, bucketNumber, ImmutableList.of(), effectivePredicate, transactionId, columnTypes);
+        this(
+                connectorId,
+                shardUuids,
+                shardDeltaMap,
+                tableSupportsDeltaDelete,
+                bucketNumber,
+                ImmutableList.of(),
+                effectivePredicate,
+                transactionId,
+                columnTypes);
     }
 
     public RaptorSplit(
             String connectorId,
             UUID shardUuid,
+            Map<UUID, UUID> shardDeltaMap,
```

Comment:
If there is only one file, we don't need a map. Just use UUID

Reply:
> If there is only one file, we don't need a map. Just use UUID

Because this kind of RaptorSplit will be converting to another structure
``
 public RaptorSplit(
            String connectorId,
            Set<UUID> shardUuids,
            Map<UUID, UUID> shardDeltaMap,
            boolean tableSupportsDeltaDelete,
            int bucketNumber,
            HostAddress address,
            TupleDomain<RaptorColumnHandle> effectivePredicate,
            OptionalLong transactionId,
            Optional<Map<String, Type>> columnTypes)
``

that's why we use a map even for only one UUID